File: Domain/Resource Projects/AtebionParse2/Cryptographer.vb
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Security.Cryptography
Imports System.Text
'' Added 02.01.2013
Public Class Cryptographer
    Private Shared _key As String

    Public Sub New()
    End Sub


    Public Shared WriteOnly Property Key() As String
        Set(ByVal value As String)
            _key = value
        End Set
    End Property

    ''' <summary>
    ''' Encrypt the given string using the default key.
    ''' </summary>
    ''' <param name="strToEncrypt">The string to be encrypted.</param>
    ''' <returns>The encrypted string.</returns>
    Public Shared Function Encrypt(ByVal strToEncrypt As String) As String
        Try
            Return Encrypt(strToEncrypt, _key)
        Catch ex As Exception
            Return "Wrong Input. " + ex.Message
        End Try

    End Function

    ''' <summary>
    ''' Decrypt the given string using the default key.
    ''' </summary>
    ''' <param name="strEncrypted">The string to be decrypted.</param>
    ''' <returns>The decrypted string.</returns>
    Public Shared Function Decrypt(ByVal strEncrypted As String) As String
        Try
            Return Decrypt(strEncrypted, _key)
        Catch ex As Exception
            Return "Wrong Input. " + ex.Message
        End Try
    End Function

    ''' <summary>
    ''' Encrypt the given string using the specified key.
    ''' </summary>
    ''' <param name="strToEncrypt">The string to be encrypted.</param>
    ''' <param name="strKey">The encryption key.</param>
    ''' <returns>The encrypted string.</returns>
    Public Shared Function Encrypt(ByVal strToEncrypt As String, ByVal strKey As String) As String
        Try
            Dim objDESCrypto As New TripleDESCryptoServiceProvider()
            Dim objHashMD5 As New MD5CryptoServiceProvider()

            Dim byteHash As Byte(), byteBuff As Byte()
            Dim strTempKey As String = strKey

            byteHash = objHashMD5.ComputeHash(ASCIIEncoding.ASCII.GetBytes(strTempKey))
            objHashMD5 = Nothing
            objDESCrypto.Key = byteHash
            objDESCrypto.Mode = CipherMode.ECB
            'CBC, CFB
            byteBuff = ASCIIEncoding.ASCII.GetBytes(strToEncrypt)
            Return Convert.ToBase64String(objDESCrypto.CreateEncryptor().TransformFinalBlock(byteBuff, 0, byteBuff.Length))
        Catch ex As Exception
            Return "Wrong Input. " + ex.Message
        End Try
    End Function


    ''' <summary>
    ''' Decrypt the given string using the specified key.
    ''' </summary>
    ''' <param name="strEncrypted">The string to be decrypted.</param>
    ''' <param name="strKey">The decryption key.</param>
    ''' <returns>The decrypted string.</returns>
    Public Shared Function Decrypt(ByVal strEncrypted As String, ByVal strKey As String) As String
        Try
            Dim objDESCrypto As New TripleDESCryptoServiceProvider()
            Dim objHashMD5 As New MD5CryptoServiceProvider()

            Dim byteHash As Byte(), byteBuff As Byte()
            Dim strTempKey As String = strKey

            byteHash = objHashMD5.ComputeHash(ASCIIEncoding.ASCII.GetBytes(strTempKey))
            objHashMD5 = Nothing
            objDESCrypto.Key = byteHash
            objDESCrypto.Mode = CipherMode.ECB
            'CBC, CFB
            byteBuff = Convert.FromBase64String(strEncrypted)
            Dim strDecrypted As String = ASCIIEncoding.ASCII.GetString(objDESCrypto.CreateDecryptor().TransformFinalBlock(byteBuff, 0, byteBuff.Length))
            objDESCrypto = Nothing

            Return strDecrypted
        Catch ex As Exception
            Return "Wrong Input. " + ex.Message
        End Try
    End Function

End Class

File: Domain/Parse/Cryptographer.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace Domain.Parse
{
  public class Cryptographer
  {
    private static string _key;

    public static string Key
    {
      set => Cryptographer._key = value;
    }

    /// <summary>Encrypt the given string using the default key.</summary>
    /// <param name="strToEncrypt">The string to be encrypted.</param>
    /// <returns>The encrypted string.</returns>
    public static string Encrypt(string strToEncrypt)
    {
      string str;
      try
      {
        str = Cryptographer.Encrypt(strToEncrypt, Cryptographer._key);
      }
      catch (Exception ex)
      {
        str = "Wrong Input. " + ex.Message;
      }
      return str;
    }

    /// <summary>Decrypt the given string using the default key.</summary>
    /// <param name="strEncrypted">The string to be decrypted.</param>
    /// <returns>The decrypted string.</returns>
    public static string Decrypt(string strEncrypted)
    {
      string str;
      try
      {
        str = Cryptographer.Decrypt(strEncrypted, Cryptographer._key);
      }
      catch (Exception ex)
      {
        str = "Wrong Input. " + ex.Message;
      }
      return str;
    }

    /// <summary>Encrypt the given string using the specified key.</summary>
    /// <param name="strToEncrypt">The string to be encrypted.</param>
    /// <param name="strKey">The encryption key.</param>
    /// <returns>The encrypted string.</returns>
    public static string Encrypt(string strToEncrypt, string strKey)
    {
      string str;
      try
      {
         using (TripleDESCryptoServiceProvider cryptoServiceProvider = new TripleDESCryptoServiceProvider())
         using (MD5CryptoServiceProvider hasher = new MD5CryptoServiceProvider())
         {
            byte[] hash = hasher.ComputeHash( Encoding.ASCII.GetBytes( strKey ) );
            cryptoServiceProvider.Key = hash;
            cryptoServiceProvider.Mode = CipherMode.ECB;
            byte[] bytes = Encoding.ASCII.GetBytes( strToEncrypt );
            str = Convert.ToBase64String( cryptoServiceProvider.CreateEncryptor().TransformFinalBlock( bytes, 0, bytes.Length ) );
         }
      }
      catch (Exception ex)
      {
        str = "Wrong Input. " + ex.Message;
      }
      return str;
    }

    /// <summary>Decrypt the given string using the specified key.</summary>
    /// <param name="strEncrypted">The string to be decrypted.</param>
    /// <param name="strKey">The decryption key.</param>
    /// <returns>The decrypted string.</returns>
    public static string Decrypt(string strEncrypted, string strKey)
    {
      string str;
      try
      {
         using(TripleDESCryptoServiceProvider cryptoServiceProvider = new TripleDESCryptoServiceProvider())
         using(MD5CryptoServiceProvider hasher = new MD5CryptoServiceProvider())
         {
           byte[] hash = hasher.ComputeHash(Encoding.ASCII.GetBytes(strKey));
           cryptoServiceProvider.Key = hash;
           cryptoServiceProvider.Mode = CipherMode.ECB;
           byte[] inputBuffer = Convert.FromBase64String(strEncrypted);
           str = Encoding.ASCII.GetString(cryptoServiceProvider.CreateDecryptor().TransformFinalBlock(inputBuffer, 0, inputBuffer.Length));
			}
      }
      catch (Exception ex)
      {
        str = "Wrong Input. " + ex.Message;
      }
      return str;
    }
  }
}
